api/agentpb/agent.proto

```diff
@@ -167,6 +167,12 @@ message StartActionRequest {
   // PTSummaryParams describes parameters for PT summary.
   message PTSummaryParams {}
 
+  // PTMySQLSummaryParams describes parameters for PT MySQL summary.
+  message PTMySQLSummaryParams {
+    string address = 1;
+    string username = 2;
+    string password = 3;
```

Comment:
Why we need address, username, and password instead of DSN as other actions above?

Reply:
The pt-mysql-summary execution requires those params.

Reply:
Okay, but can we extract them from DSN?

Reply:
It takes them from DSN.

Reply:
Who is "it"?
pt-mysql-summary does accept address – it accepts host and port, and we have to parse address anyway. So let's pass  dsn for consistency with other actions above and for socket support.

Reply:
@AlekSi, For agents we decide on pmm-managed side what flags should be passed to processes, why do we let pmm-agent to decide how to run process? We can send list of flags like for exporters or we can send address and port separately since we store them separately in our db. So what's the reason to generate dsn and parse it later? Do we expect that pt-mysql-summary will accept dsn with additional query parameters in future?

Reply:
> what flags should be passed to processes

For Agents, not for Actions. We don't send CLI flags with dashes there, we send parameters.

I think we should pass DNS because we use DSN for all other Actions there. In the current implementation, we have to do parameters parsing anyway (split address into host and port), and socket support is missing.

Reply:
I don't suggest to use current implementation. I'm suggesting to use separate fields for socket, host and port to not overcomplicate code on pmm-agent side.
The only reason why we should use dsn is possible backward compatibility in future if we decide to add some query parameters, are we going to implement that later?

Reply:
@yareach Discussed that with @BupycHuk. Ok, let's pass separate fields for address (what pt-mysql-summary calls host), port, UNIX socket (that was missing, AFAIU), username and password